cassiopeia/solver/MaxCutGreedySolver.py

```diff
@@ -92,51 +95,88 @@ def perform_split(
         Returns:
             A tuple of lists, representing the left and right partitions
         """
-        freq = 0
-        char = 0
-        state = ""
-        for i in mutation_frequencies:
-            for j in mutation_frequencies[i]:
-                if j != self.missing_char and j != "0":
+        best_frequency = 0
+        chosen_character = 0
+        chosen_state = 0
+        for character in mutation_frequencies:
+            for state in mutation_frequencies[character]:
+                if state != self.missing_char and state != 0:
                     # Avoid splitting on mutations shared by all samples
                     if (
-                        mutation_frequencies[i][j] > freq
-                        and mutation_frequencies[i][j]
+                        mutation_frequencies[character][state]
                         < len(samples)
-                        - mutation_frequencies[i][self.missing_char]
+                        - mutation_frequencies[character][self.missing_char]
                     ):
-                        char, state = i, j
-                        freq = mutation_frequencies[i][j]
-
-        if state == "":
+                        if self.weights:
+                            if (
+                                mutation_frequencies[character][state]
+                                * self.weights[character][state]
+                                > best_frequency
+                            ):
+                                chosen_character, chosen_state = (
+                                    character,
+                                    state,
+                                )
+                                best_frequency = (
+                                    mutation_frequencies[character][state]
+                                    * self.weights[character][state]
+                                )
+                        else:
+                            if (
+                                mutation_frequencies[character][state]
+                                > best_frequency
+                            ):
+                                chosen_character, chosen_state = (
+                                    character,
+                                    state,
+                                )
+                                best_frequency = mutation_frequencies[
+                                    character
+                                ][state]
+
+        if chosen_state == 0:
             return samples, []
 
         left_set = []
         right_set = []
         missing = []
 
         for i in samples:
-            if self.prune_cm.iloc[i, char] == state:
+            if (
+                self.unique_character_matrix.loc[i, :][chosen_character]
```

Comment:
Why not index like this: `self.unique_character_matrix.loc[i, chosen_character]`?

Reply:
this actually throws an error if the column indices aren't explicitly integers, so this way allows you to use the indexing of the row by index and the column by integer

Reply:
kind of like a hybrid of loc for row and iloc for column

Reply:
Sounds good, thanks for answering!